crates/yttrium/src/chain_abstraction/api/route.rs

```diff
@@ -33,8 +34,34 @@ pub struct FundingMetadata {
     pub chain_id: String,
     pub token_contract: Address,
     pub symbol: String,
+
+    // The amount that was sourced (includes the bridging fee)
     pub amount: U256,
+
+    // The amount taken by the bridge as a fee
     pub bridging_fee: U256,
+
+    #[serde(
+        deserialize_with = "crate::utils::deserialize_unit",
+        serialize_with = "crate::utils::serialize_unit"
+    )]
+    #[serde(default = "default_unit")]
+    pub decimals: Unit,
```

Comment:
I believe we can use U8 here instead?

Reply:
Why? I think we should use `Unit` since it's alloy's type to represent these decimals. It has built-in assertions that the value is less than 77, so we know it's a valid unit. Rust code can also have fewer error conditions when this type is already being used.